lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,306 @@
+import { createHash } from 'crypto';
+import readline from 'readline';
+import { createUnzip } from 'zlib';
+import type { GetReleasesConfig, ReleaseResult } from '..';
+import { logger } from '../../logger';
+import * as fs from '../../util/fs';
+import type { HttpOptions } from '../../util/http';
+import { Datasource } from '../datasource';
+import type { DebLanguageConfig, PackageDescription } from './types';
+
+/**
+ * @brief This datasource allows to download
+ */
+export class DebDatasource extends Datasource {
+  static readonly id = 'deb';
+
+  constructor() {
+    super(DebDatasource.id);
+  }
+
+  /**
+   * This is just an internal list of compressions that are supported and tried to be downloaded from the remote
+   */
+  static readonly compressions = ['gz'];
+
+  /**
+   * Users are able to specify custom Debian repositories as long as they follow
+   * the Debian package repository format as specified here
+   * @see{https://wiki.debian.org/DebianRepository/Format}
+   */
+  override readonly customRegistrySupport = true;
+
+  /**
+   * The original apt source list file format is
+   * deb uri distribution [component1] [component2] [...]
+   * @see{https://wiki.debian.org/DebianRepository/Format}
+   *
+   * However, for Renovate, we require the registry URLs to be
+   * valid URLs which is why the parameters are encoded in the URL.
+   *
+   * The following query parameters are required:
+   * - components: comma separated list of components
+   * - suite: stable, oldstable or other alias for a release, either this or release must be given
+   * - release: buster, etc.
+   */
+  override readonly defaultRegistryUrls = [
+    'https://ftp.debian.org/debian?suite=stable&components=main,contrib,non-free',
+  ];
+
+  override readonly defaultConfig: DebLanguageConfig = {
+    deb: {
+      binaryArch: 'amd64',
```

Comment:
Why now add this to registry url too. Make it optional and fallback to `amd64` by default?

Reply:
good idea, I'll then rename the current binary arch configuration to "defaultBinaryArch"